### PR TITLE
BUG: value_count(normalize=True, dropna=True) counting missing in denominator, #12558 

### LINKED ISSUE #12558: BUG?: value_counts(normalize=True) normalizes over all observations including NaN.
By default, `value_counts` ignores missing values. To get them displayed, one must add the option `dropna=False`. But when one normalizes, they are entering into the denominator even when `dropna=True`. 

#### Code Sample, a copy-pastable example if possible

    s = pd.Series([1,2,3,np.nan, np.nan, np.nan])
    s.value_counts(normalize=True)

    0.1667
    0.1667
    0.1667

#### Expected Output

    0.333
    0.333
    0.333

#### output of ``pd.show_versions()``

pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.4.4.final.0
python-bits: 64
OS: Darwin
OS-release: 15.3.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.7
pip: 8.1.0
setuptools: 20.2.2
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.16.1
statsmodels: None
IPython: 4.0.1
sphinx: 1.3.1
patsy: 0.4.0
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.4.4
matplotlib: 1.4.3
openpyxl: 2.2.6
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.7.7
lxml: 3.4.4
bs4: 4.4.1
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 1.0.9
pymysql: None
psycopg2: None
Jinja2: 2.8

### PR DESCRIPTION
**Pull request by nickeubank**
Closes #12558

### CODE REVIEW

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -342,7 +343,10 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
         result = result.sort_values(ascending=ascending)
 
     if normalize:
-        result = result / float(values.size)
+        if dropna:
+            result = result / float(values[notnull(values)].size)
```
Comment (jreback) [inline pandas/core/algorithms.py]: ``Series(values).count()``

File: pandas/tests/test_algos.py [hunk 2]
```diff
@@ -517,6 +517,12 @@ def test_dropna(self):
                 pd.Series([10.3, 5., 5., None]).value_counts(dropna=False),
                 pd.Series([2, 1, 1], index=[5., 10.3, np.nan]))
 
+    def test_dropna_normalize(self):
+        # Issue 12558
+        tm.assert_series_equal(
+            pd.Series([ 5.,10.3,10.3,10.3,np.nan]).value_counts(dropna=True, normalize=True),
```
Comment (jreback) [inline pandas/tests/test_algos.py]: do all 4 combos here (``dropna,normalize``)

Comment (jreback) [inline pandas/tests/test_algos.py]: just call this ``test_normalize``

Comment (jreback) [inline pandas/tests/test_algos.py]: add a test that uses ``NaT`` as well (e.g. datetimes)

**Comment by kawochen** [comment]
#12566 fixes this too, will need to fix groupby. Lemme know if you wanna work on this and I'll close mine.

**Comment by nickeubank** [comment]
@kawochen Oops! Sorry, did not mean to double up. If you have this covered I can close -- not sure what you mean about groupby.

**Comment by kawochen** [comment]
`groupby` has a version of `value_counts`.  I fixed it for `numpy` > 1.8 and have an idea for older versions.  Lemme play around with it tonight (EST).  If it doesn't work I'll close it and let you have fun :)

**Pull request closed without merging**